### PR TITLE
Add missing build step to boost formula

### PR DESCRIPTION
**Pull request by aw1621107**
See: http://lists.boost.org/boost-users/2014/12/83414.php and followup

--HEAD --with-mpi --without-single builds were failing with a missing
header sometime when compiling the threads component of the library.
Turns out another command was added to the build sequence, but the docs
weren't updated. Now, instead of

  ./bootstrap.sh
  ./b2

the proper build commands should be

  ./bootstrap.sh
  ./b2 headers
  ./b2

### CODE REVIEW

**Comment by aw1621107** [comment]
```
==> brew uses boost
Error: undefined method `has_apple_developer_tools?' for OS::Mac:Module
```
Well that's fun... Anything I could do about that?

**Comment by DomT4** [comment]
@aw1621107 You may need to ensure the Homebrew your branch was based off of is completely up-to-date.

**Comment by aw1621107** [comment]
Alright, updated, I think.

If the merge commit is a problem I can resubmit a clean PR. Just thought since PRs are not directly merged that the merge commit wouldn't be much of an issue, and it lets me keep using this PR instead of making a dupe. If not, let me know.

File: Library/Formula/boost.rb [hunk 1]
```diff
@@ -119,6 +119,7 @@ def install
     end
 
     system "./bootstrap.sh", *bootstrap_args
+    system "./b2 headers"
```
Comment (mikemcquaid) [inline Library/Formula/boost.rb]: Should be `system "./b2", "headers"`

Comment (mikemcquaid) [inline Library/Formula/boost.rb]: Actually, I'll just fix it up locally.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `boost: add missing build step.`.

**Comment by aw1621107** [comment]
Should I still split up the "./b2 headers" line? Doesn't seem it's been split in the official repo

**Comment by DomT4** [comment]
I'll fix it in a sec.

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew/commit/93d7446483660f375271a469a09cb485b13ef4c2

**Pull request closed without merging**